### ISSUE TITLE
Change in 1.2.0 breaks some const usage

### ISSUE THREAD

**Issue by MWumpusZ** [funkwerk/accessors#19]
Using 1.1.1, this worked:
```
import accessors;

struct Thing
{
    @Read
    private int[] content_;

    mixin(GenerateFieldAccessors);
}

class User
{
    void helper(const int[] content)
    {
    }

    void doer(const Thing thing)
    {
        helper(thing.content);
    }
}
```

In 1.2.0 it no longer does:
`Test.d(19): Error: mutable method Test.Thing.content is not callable using a const object`

@duselbaer points at this commit as a possible suspect: https://github.com/funkwerk/accessors/commit/b13f84a479e62937ac65079c197ea1c5bd396575 - the `inout` appears to have gotten lost?

**Comment by MWumpusZ** [comment]
On a second look: that's ALL that that commit does - so it's obviously intentional.  But is it not the cause for the above no longer working?

**Comment by MWumpusZ** [comment]
Using `@ConstRead` (instead of `@Read`) works and is possible in my particular case, but doesn't seem like a general solution? :(

**Comment by belka-ew** [comment]
The problem with `inout` is that if an `inout`-function returns a reference type (class, dynamic array) it returns it as `const`. And it isn't what we want for `@Read`. `@Read` is supposed to return a mutable reference. Therefore `inout` was removed from the properties generated by `@Read` (See the unittest in the referenced commit for an example). But it makes also your code not working since a mutable property is called on a const object.
Is there a problem with using both `@Read` and `@ConstRead` together? It would generate both properties for mutable and const objects and the appropriate one would be called.

**Comment by MWumpusZ** [comment]
I...  guess not?  At first glance it looks a bit odd?
In the specific case where it came up, I was able to switch to `ConstRead` (only) this morning anyway.

**Comment by duselbaer** [comment]
Puh, is that the way inout is supposed to work? Is it possible that @Read generates both - const and mutable by default? I really don't like writing @ConstRead everywhere. The mutable @Read seems to be a workaround for the old value object implementations which just don't care about const-nesss.

So:
@Read --> shall create a const function - maybe both for downwards compatibility?
@MutableRead --> non-const?

**Comment by belka-ew** [comment]
I was wrong, `inout`-functions return a mutable reference if the return type is marked as `inout` what wasn't the case in the original implementation. Maybe `inout` should indeed come back together with `inout` on the return type. I have to check if all tests pass then.

**Comment by belka-ew** [comment]
Fixed in 4df5c353931cf13fb97cf92157cd7d8843d0e442.
